packages/ckeditor5-html-support/src/integrations/dualcontent.ts

```diff
@@ -32,59 +34,59 @@ import DataFilter from '../datafilter';
  * {@link module:html-support/dataschema~DataSchemaDefinition#modelSchema} in editor schema.
  * Otherwise, it will be registered under {@link module:html-support/dataschema~DataSchemaBlockElementDefinition#paragraphLikeModel} model
  * name with model schema accepting only inline content (inheriting from `$block`).
- *
- * @extends module:core/plugin~Plugin
  */
 export default class DualContentModelElementSupport extends Plugin {
 	/**
 	 * @inheritDoc
 	 */
-	static get requires() {
+	public static get requires(): PluginDependencies {
 		return [ DataFilter ];
 	}
 
 	/**
 	 * @inheritDoc
 	 */
-	static get pluginName() {
+	public static get pluginName(): 'DualContentModelElementSupport' {
 		return 'DualContentModelElementSupport';
 	}
 
 	/**
 	 * @inheritDoc
 	 */
-	init() {
+	public init(): void {
 		const dataFilter = this.editor.plugins.get( DataFilter );
 
-		dataFilter.on( 'register', ( evt, definition ) => {
+		dataFilter.on<RegisterEvent>( 'register', ( evt, definition ) => {
+			const blockDefinition = definition as DataSchemaBlockElementDefinition;
 			const editor = this.editor;
 			const schema = editor.model.schema;
 			const conversion = editor.conversion;
 
-			if ( !definition.paragraphLikeModel ) {
+			if ( !blockDefinition.paragraphLikeModel ) {
 				return;
 			}
 
 			// Can only apply to newly registered features.
-			if ( schema.isRegistered( definition.model ) || schema.isRegistered( definition.paragraphLikeModel ) ) {
+			if ( schema.isRegistered( blockDefinition.model ) || schema.isRegistered( blockDefinition.paragraphLikeModel ) ) {
 				return;
 			}
 
-			const paragraphLikeModelDefinition = {
-				model: definition.paragraphLikeModel,
-				view: definition.view
+			const paragraphLikeModelDefinition: DataSchemaBlockElementDefinition = {
+				model: blockDefinition.paragraphLikeModel,
+				view: blockDefinition.view,
+				isBlock: true
```

Comment:
```suggestion
                view: blockDefinition.view
```

I'd make `DataSchemaBlockElementDefinition#isBlock` optional.